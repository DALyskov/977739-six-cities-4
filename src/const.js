export const APPROVED_NAME = [
  `Beautiful & luxurious apartment at great location`,
  `Wood and stone place`,
  `Canal View Prinsengracht`,
  `Nice, cozy, warm big bed apartment`,
];
