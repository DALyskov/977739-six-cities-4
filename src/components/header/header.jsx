import React from 'react';
import propTypes from 'prop-types';
import {connect} from 'react-redux';

import {AuthorizationStatus, HeaderClassNames, PageType} from '../../const.js';

import {ActionCreator as AppActionCreator} from '../../reducer/state-application/state-application.js';
import {getActivePage} from '../../reducer/state-application/selectors.js';
import {
  getAuthorizationStatus,
  getUserEmail,
} from '../../reducer/user/selectors.js';

const Header = (props) => {
  const {
    authorizationStatus,
    userEmail,
    activePage,
    onLoginClick,
    onLogoClick,
  } = props;

  const isLoggedIn = authorizationStatus === AuthorizationStatus.AUTH;

  let isActiveLink = true;
  let className = HeaderClassNames.OTHER_PAGE;

  if (activePage === PageType.MAIN) {
    isActiveLink = false;
    className = HeaderClassNames.MAIN;
  }

  return (
    <header className="header">
      <div className="container">
        <div className="header__wrapper">
          <div className="header__left">
            {isActiveLink ? (
              <a
                className={className}
                href="main.html"
                onClick={(evt) => {
                  evt.preventDefault();
                  onLogoClick();
                }}>
                <img
                  className="header__logo"
                  src="img/logo.svg"
                  alt="6 cities logo"
                  width="81"
                  height="41"
                />
              </a>
            ) : (
              <a className={className}>
                <img
                  className="header__logo"
                  src="img/logo.svg"
                  alt="6 cities logo"
                  width="81"
                  height="41"
                />
              </a>
            )}
          </div>
          <nav className="header__nav">
            <ul className="header__nav-list">
              <li className="header__nav-item user">
                <a
                  className="header__nav-link header__nav-link--profile"
                  href="#"
                  onClick={(evt) => {
                    evt.preventDefault();
                    onLoginClick();
                  }}>
                  <div className="header__avatar-wrapper user__avatar-wrapper"></div>
                  {isLoggedIn ? (
                    <span className="header__user-name user__name">
                      {userEmail}
                    </span>
                  ) : (
                    <span className="header__login">Sign in</span>
                  )}
                </a>
              </li>
            </ul>
          </nav>
        </div>
      </div>
    </header>
  );
};

Header.propTypes = {
  authorizationStatus: propTypes.oneOf(Object.values(AuthorizationStatus))
    .isRequired,
  userEmail: propTypes.oneOfType([propTypes.string, propTypes.bool]).isRequired,
  activePage: propTypes.oneOf(Object.values(PageType)).isRequired,
  onLoginClick: propTypes.func.isRequired,
  onLogoClick: propTypes.func.isRequired,
};

const mapStateToProps = (state) => ({
  authorizationStatus: getAuthorizationStatus(state),
  userEmail: getUserEmail(state),
  activePage: getActivePage(state),
});

const mapDispatchToProps = (dispatch) => ({
  onLoginClick() {
    dispatch(AppActionCreator.changeActivePage(PageType.SING_IN));
  },
  onLogoClick() {
    dispatch(AppActionCreator.changeActivePage(PageType.MAIN));
  },
});

export {Header};
export default connect(mapStateToProps, mapDispatchToProps)(Header);
