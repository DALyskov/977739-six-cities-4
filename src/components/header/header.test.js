import React from 'react';
import renderer from 'react-test-renderer';

import {AuthorizationStatus, PageType} from '../../const.js';

import {Header} from './header.jsx';

describe(`Header_snapchots`, () => {
  it(`is_logged_in_for_main`, () => {
    const tree = renderer
      .create(
        <Header
          authorizationStatus={AuthorizationStatus.AUTH}
          userEmail={`A@mail.com`}
          activePage={PageType.MAIN}
          onLoginClick={() => {}}
          onLogoClick={() => {}}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it(`is_not_logged_in_for_main`, () => {
    const tree = renderer
      .create(
        <Header
          authorizationStatus={AuthorizationStatus.NO_AUTH}
          userEmail={false}
          activePage={PageType.MAIN}
          onLoginClick={() => {}}
          onLogoClick={() => {}}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it(`is_logged_in_for_other_page`, () => {
    const tree = renderer
      .create(
        <Header
          authorizationStatus={AuthorizationStatus.AUTH}
          userEmail={`A@mail.com`}
          activePage={PageType.PROPERTY}
          onLoginClick={() => {}}
          onLogoClick={() => {}}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it(`is_not_logged_in_for_other_page`, () => {
    const tree = renderer
      .create(
        <Header
          authorizationStatus={AuthorizationStatus.NO_AUTH}
          userEmail={false}
          activePage={PageType.PROPERTY}
          onLoginClick={() => {}}
          onLogoClick={() => {}}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
