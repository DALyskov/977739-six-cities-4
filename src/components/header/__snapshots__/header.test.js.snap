// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header_snapchots is_logged_in_for_main 1`] = `
<header
  className="header"
>
  <div
    className="container"
  >
    <div
      className="header__wrapper"
    >
      <div
        className="header__left"
      >
        <a
          className="header__logo-link"
        >
          <img
            alt="6 cities logo"
            className="header__logo"
            height="41"
            src="/img/logo.svg"
            width="81"
          />
        </a>
      </div>
      <nav
        className="header__nav"
      >
        <ul
          className="header__nav-list"
        >
          <li
            className="header__nav-item user"
          >
            <a
              className="header__nav-link header__nav-link--profile"
              href="/favorites"
              onClick={[Function]}
            >
              <div
                className="header__avatar-wrapper user__avatar-wrapper"
              />
              <span
                className="header__user-name user__name"
              >
                A@mail.com
              </span>
            </a>
          </li>
        </ul>
      </nav>
    </div>
  </div>
</header>
`;

exports[`Header_snapchots is_logged_in_for_other_page 1`] = `
<header
  className="header"
>
  <div
    className="container"
  >
    <div
      className="header__wrapper"
    >
      <div
        className="header__left"
      >
        <a
          className="header__logo-link header__logo-link--active"
          href="/"
          onClick={[Function]}
        >
          <img
            alt="6 cities logo"
            className="header__logo"
            height="41"
            src="/img/logo.svg"
            width="81"
          />
        </a>
      </div>
      <nav
        className="header__nav"
      >
        <ul
          className="header__nav-list"
        >
          <li
            className="header__nav-item user"
          >
            <a
              className="header__nav-link header__nav-link--profile"
              href="/favorites"
              onClick={[Function]}
            >
              <div
                className="header__avatar-wrapper user__avatar-wrapper"
              />
              <span
                className="header__user-name user__name"
              >
                A@mail.com
              </span>
            </a>
          </li>
        </ul>
      </nav>
    </div>
  </div>
</header>
`;

exports[`Header_snapchots is_not_logged_in_for_main 1`] = `
<header
  className="header"
>
  <div
    className="container"
  >
    <div
      className="header__wrapper"
    >
      <div
        className="header__left"
      >
        <a
          className="header__logo-link"
        >
          <img
            alt="6 cities logo"
            className="header__logo"
            height="41"
            src="/img/logo.svg"
            width="81"
          />
        </a>
      </div>
      <nav
        className="header__nav"
      >
        <ul
          className="header__nav-list"
        >
          <li
            className="header__nav-item user"
          >
            <a
              className="header__nav-link header__nav-link--profile"
              href="/login"
              onClick={[Function]}
            >
              <div
                className="header__avatar-wrapper user__avatar-wrapper"
              />
              <span
                className="header__login"
              >
                Sign in
              </span>
            </a>
          </li>
        </ul>
      </nav>
    </div>
  </div>
</header>
`;

exports[`Header_snapchots is_not_logged_in_for_other_page 1`] = `
<header
  className="header"
>
  <div
    className="container"
  >
    <div
      className="header__wrapper"
    >
      <div
        className="header__left"
      >
        <a
          className="header__logo-link header__logo-link--active"
          href="/"
          onClick={[Function]}
        >
          <img
            alt="6 cities logo"
            className="header__logo"
            height="41"
            src="/img/logo.svg"
            width="81"
          />
        </a>
      </div>
      <nav
        className="header__nav"
      >
        <ul
          className="header__nav-list"
        >
          <li
            className="header__nav-item user"
          >
            <a
              className="header__nav-link header__nav-link--profile"
              href="/login"
              onClick={[Function]}
            >
              <div
                className="header__avatar-wrapper user__avatar-wrapper"
              />
              <span
                className="header__login"
              >
                Sign in
              </span>
            </a>
          </li>
        </ul>
      </nav>
    </div>
    <div
      className="err-message"
      style={
        Object {
          "color": "red",
          "margin": "10px auto",
          "textAlign": "center",
        }
      }
    >
      abcd
    </div>
  </div>
</header>
`;
